pkg/app/instances/docker.go

```diff
@@ -53,11 +58,63 @@ const (
 	DeleteHostOPType OPType = "deletehost"
 )
 
-func NewDockerInstanceManager(cfg Config, cli *client.Client) *DockerInstanceManager {
-	return &DockerInstanceManager{
-		Config: cfg,
-		Client: cli,
+func NewDockerInstanceManager(cfg Config, cli *client.Client) (*DockerInstanceManager, error) {
+	ctx := context.TODO()
```

Comment:
The new logic to manage images should be when creating the hosts (docker containers) rather than when creating an instance of the `DockerInstanceManager`.

Reply:
Could you elaborate about why you prefer this way? IMO, docker instance manager without any docker image cannot work its role properly.

Reply:
It's about consistency with other instance manager implementations like GCP, if we use an invalid GCP image resource name, we are still able to launch the cloud orchestrator binary,  we fail  with error `500 Internal Server Error` Image Not Found, but we don't prevent the Cloud Orchestrator binary to start running. 

Making this logic part of the constructor makes the binary to crash at the beginning, the service won't ever start. If we restart the orchestrator in a machine that already have hosts running in a new version deployment, by having any issue in this new logic in the constructor the binary will crash and never start, hence we cannot even list and connect with the existing hosts, we should only be affecting the create logic, but not the whole service.

Reply:
Understood, and it sounds fair. I wasn't aware of such history on GCE impl.

Reply:
Done.